deepchecks/checks/performance/regression_error_distribution.py

```diff
@@ -0,0 +1,89 @@
+
+    def run(self, dataset: Dataset, model: BaseEstimator) -> CheckResult:
+        """Run check.
+
+        Arguments:
+            dataset (Dataset): A dataset object.
+            model (BaseEstimator): A scikit-learn-compatible fitted estimator instance
+
+        Returns:
+           CheckResult:
+                - value is a dict with mse and kurtosis values.
+                - display is histogram of error distirbution and the extreme prediction errors.
+
+        Raises:
+            DeepchecksValueError: If the object is not a Dataset instance with a label
+        """
+        return self._regression_error_distribution(dataset, model)
+
+    def _regression_error_distribution(self, dataset: Dataset, model: BaseEstimator):
+        check_name = self.__class__.__name__
+        Dataset.validate_dataset(dataset, check_name)
+        dataset.validate_label(check_name)
+        task_type_validation(model, dataset, [ModelType.REGRESSION], check_name)
+
+        y_test = dataset.label_col()
+        y_pred = model.predict(dataset.features_columns())
+
+        mse = mean_squared_error(dataset.label_col(), y_pred, squared=False)
+        diff = y_test - y_pred
+        kurtosis_value = kurtosis(diff)
+
+        n_largest_diff = diff.nlargest(self.n_extreme)
+        n_largest_diff.name= n_largest_diff.name + '_pred_diff'
+        n_largest = pd.concat([dataset.data.loc[n_largest_diff.index], n_largest_diff], axis=1)
+
+        n_smallest_diff = diff.nsmallest(self.n_extreme)
+        n_smallest_diff.name= n_smallest_diff.name + '_pred_diff'
+        n_smallest = pd.concat([dataset.data.loc[n_smallest_diff.index], n_smallest_diff], axis=1)
+
+        def display_hist():
+            diff = y_test - y_pred
+            diff.hist(bins = 40)
+            plt.title('Histogram of prediction errors')
+            plt.xlabel(f'{dataset.label_name()} prediction error')
+            plt.ylabel('Frequency')
+
+        display = [display_hist, 'Largest over estimation errors:', n_largest,
+                   'Largest under estimation errors:', n_smallest,]
+        return CheckResult(value={'mse': mse, 'kurtosis': kurtosis_value}, check=self.__class__, display=display)
+
+    def add_condition_absolute_kurtosis_not_greater_than(self, max_kurtosis: float = 0.1):
+        """Add condition - require the absolute kurtosis value to not surpass max_kurtosis.
+
+        Args:
+            max_kurtosis (float): Maximum absolute kurtosis value
+        """
+        def max_kurtosis_condition(result: float) -> ConditionResult:
+            kurtosis_value = result['kurtosis']
+            if abs(kurtosis_value) > max_kurtosis:
+                return ConditionResult(False, f'kurtosis: {format_number(kurtosis_value, 5)}')
+            else:
+                return ConditionResult(True)
+
+        return self.add_condition(f'Absolute kurtosis value is not greater than {format_number(max_kurtosis, 5)}',
+                                  max_kurtosis_condition)
```

Comment:
Why no condition on MSE?

Reply:
it's a bit hard to write a condition for it as the value too dependent on the data

Reply:
maybe we don't really need to calculate it here even, we have a better usage of it in regression bias and we actually use it in a condition. what do you think? I think I should remove it

Reply:
why is it too dependent on the data? it's an averaged value

Reply:
an error of 3000 can be big for some data and tiny for other data

Reply:
I agree its dependent on the data, and that's exactly why it should be a condition. Conditions are there for exactly this purpose, in my opinion. 
Also, would do this as a condition on average RMSE and not the most extreme sample.

Reply:
average RMSE? 
and still, wouldn't it be better to just remove it. it sounds more like the bias check than distribution check

Reply:
Meant RMSE and not average RMSE. Meant not to check the largest error but the average one.
@noamzbr WDYT?

Reply:
We already have that functionality (condition on metric) in PerformanceReport, so seems kinda redundant to have it here